build_metal.sh

```diff
@@ -1,67 +1,31 @@
-    - Caveats:
-        - At least one of these folders has to be named `build`, and if using tracy config, it has to be named `build`.. pending issue #8767.
-        - They have to be built with the original build folder name, i.e., you cannot change the build folder name after building because it will mess up the RPATHs and linking.
-
-Example:
-    ./create_venv.sh
-    cmake -B build -G Ninja && ninja -C build                       # <- Build in Release, inside folder called `build`.
-    cmake -DCMAKE_BUILD_TYPE=Debug -B build_debug -G Ninja && ninja -C build_debug    # <- Build in Debug, inside folder called `build_debug`.
-    source python_env/bin/activate                                  # <- You cannot run pytests yet since pybinds have not been installed.
-    ninja install -C build                                          # <- Install Release pybinds.
-    <run a pytest>                                                  # <- This test ran in Release config.
-    ninja install -C build_debug                                    # <- Install Debug pybinds.
-    <run a pytest>                                                  # <- This test ran in Debug config.
-
-NOTE ON DEBUGGING!:
-    GDB/LLDB is not stable right now. Recommend using GCC11 or higher for debugging or Clang-17 with GDB 14+.
-'
 
 set -eo pipefail
 
 # Function to display help
 show_help() {
-    echo "Usage: $0 [-h] [-e] [-c] [-b build_type] [-t] [-a]"
-    echo "  -h  Show this help message."
-    echo "  -e  Enable CMAKE_EXPORT_COMPILE_COMMANDS."
-    echo "  -c  Enable ccache for the build."
-    echo "  -b  Set the build type. Default is Release. Other options are Debug, RelWithDebInfo, and CI."
-    echo "  -t  Enable build time trace (clang only)."
-    echo "  -a  Enable AddressSanitizer."
-    echo "  -m  Enable MemorySanitizer."
-    echo "  -s  Enable ThreadSanitizer."
-    echo "  -u  Enable UndefinedBehaviorSanitizer."
-    echo "  -p  Enable Tracy profiler."
+    echo "Usage: $0 [options]..."
+    echo "  -h, --help                       Show this help message."
+    echo "  -e, --export-compile-commands    Enable CMAKE_EXPORT_COMPILE_COMMANDS."
+    echo "  -c, --enable-ccache              Enable ccache for the build."
+    echo "  -b, --build-type build_type      Set the build type. Default is Release. Other options are Debug, RelWithDebInfo, and CI."
+    echo "  -t, --trace                      Enable build time trace (clang only)."
+    echo "  -a, --enable-asan                Enable AddressSanitizer."
+    echo "  -m, --enable-msan                Enable MemorySanitizer."
+    echo "  -s, --enable-tsan                Enable ThreadSanitizer."
+    echo "  -u, --enable-ubsan               Enable UndefinedBehaviorSanitizer."
+    echo "  -p, --enable-profiler            Enable Tracy profiler."
+    echo "  --install-prefix                 Where to install build artifacts."
+    echo "  --build-tests                    Build Testcases."
```

Comment:
should there be a short form for this option? This will be used often I believe

Reply:
Both `b` and `t` are taken. What would you suggest? `z`?

Reply:
main reason why I ask is because I can imagine metal devs would be annoyed by having to do all this extra typing

Reply:
I advertised this plan to the developers and didn't get any push back on `--build-tests`.
If you have a good idea for a short switch, I'll add it.
Otherwise, maybe we can wait for a complaint?

Reply:
@mo-tenstorrent See the discussion here... I am adding optional test building to this script.